Backend/starter_project_dotnetforce7/BlogApp2/BlogApp.Persistence/Repositories/UnitOfWork.cs

```diff
@@ -28,15 +29,25 @@ public UnitOfWork(BlogAppDbContext context)
             } 
          }
 
+        public IRateRepository RateRepository
+        {
+            get
+            {
+                if (_rateRepository == null)
+                    _rateRepository = new RateRepository(_context);
+                return _rateRepository;
+            }
+        }
+
         public void Dispose()
         {
             _context.Dispose();
             GC.SuppressFinalize(this);
         }
 
-        public async Task Save()
+        public async Task<int> Save()
         {
-            await _context.SaveChangesAsync();
+            return await _context.SaveChangesAsync();
```

Comment:
Make sure this change is communicated within the team and make sure it's being utilized in CQRS handlers.